storage/innobase/dict/dict0stats.cc

```diff
@@ -3623,7 +3623,8 @@ dict_stats_update_for_index(
 		}
 	}
 
-	dict_stats_update_transient_for_index(index);
+	if (index->is_btree())
+		dict_stats_update_transient_for_index(index);
```

Comment:
This could leave the statistics uninitialized, for example if a B-tree index has been flagged as corrupted. I would suggest a change at the lower level:
```diff
diff --git a/storage/innobase/dict/dict0stats.cc b/storage/innobase/dict/dict0stats.cc
index 49a604ef9f5..2f0db57440a 100644
--- a/storage/innobase/dict/dict0stats.cc
+++ b/storage/innobase/dict/dict0stats.cc
@@ -1152,6 +1152,7 @@ dict_stats_update_transient_for_index(
 		goto dummy_empty;
 #endif /* UNIV_DEBUG || UNIV_IBUF_DEBUG */
 	} else if (dict_index_is_online_ddl(index) || !index->is_committed()
+		   || !index->is_bree()
 		   || !index->table->space) {
 		goto dummy_empty;
 	} else {
```